### DISCUSSION TITLE
[Question]: Nav does not update the SelectedKey before firing OnSelect

### DISCUSSION THREAD

**Discussion by ghurlman** [microsoft/powercat-creator-kit#377] [Q&A] [answered]
**Describe the bug**
When clicking on a nav item, the SelectedKey property is not being updated until after the event fires. So, it always seems to be a click "behind". SelectedKey *is* updated properly before OnChange is fired, however.

**To Reproduce**
1. Create an app, add the Nav control, configure the Items property
2. Add `Notify("SelectedKey: " & Self.SelectedKey` to the `OnSelect` behavior property
3. Run the app in preview mode, and click a nav item.
4. Click a different nav item.

**Expected behavior**
A notification pops up, properly reflecting the item clicked, and then again with the second click

**Actual behavior**
The first notification shows a blank SelectedKey, and the second notification shows the SelectedKey for the first item clicked.

**Comment by denise-msft** [comment, marked as answer]
Thanks for reporting this @ghurlman.

Nav has been implemented in a way that is intended to bind a variable in Power Apps to the Nav.SelectedKey property. This enables a way to set a default value to display on first load state (e.g., the first item in the Nav on app load, display the current page of the app). 

This can be done by: 
- Initializing `varSelectedKey` to the desired record (e.g., in App.OnStart): `Set(varSelectedKey, First(NavItems).ItemKey);`
- Setting Nav.SelectedKey to `varSelectedKey`
- Updating `varSelectedKey` in Nav.OnChange: `Set(varSelectedKey, Self.Selected.ItemKey); Notify(varSelectedKey);`

At this point, the notification should be showing you the current value of `varSelectedKey`.

Is this acceptable? I realize it might not reflect the way in-product components are implemented, however it is consistent with how Creator Kit controls have been implemented. If so, we can update documentation to make these steps clearer.
